### DISCUSSION TITLE
binaries.prisma.sh timeout error

### DISCUSSION THREAD

**Discussion by BEDIRHANSAMSA** [prisma/prisma#19258] [Q&A]
npm ERR! command sh -c npx prisma@4.13.0 generate --schema ./prisma/schema.prisma
npm ERR! Error: request to https://binaries.prisma.sh/all_commits/1e7af066ee9cb95cf3a403c78d9aab3e6b04f37a/debian-openssl-3.0.x/libquery_engine.so.node.gz.sha256 failed, reason: Socket connection timeout

![image](https://github.com/prisma/prisma/assets/44173331/b70871a7-0a30-4c31-a5bc-c297ea6e2b39)
![image](https://github.com/prisma/prisma/assets/44173331/dd6602a0-e264-4715-b35a-624df249aafb)

any ideas to fix?

**Comment by ludralph** [comment]
Hi @BEDIRHANSAMSA 

This looks like a problem with `npm`. You might want to try the solution suggested [here](https://stackoverflow.com/questions/72442682/npm-err-code-err-socket-timeout-npm-err-network-socket-timeout)

**Reply by BEDIRHANSAMSA** [reply]
i tried but not work, look like my server blocking

**Reply by ludralph** [reply]
Are you using a vpn?

**Reply by BEDIRHANSAMSA** [reply]
no i'm not using

**Reply by BEDIRHANSAMSA** [reply]
It works when I try with curl but not when I use wget
```curl
curl https://binaries.prisma.sh/all_commits/d9a4c5988f480fa576d43970d5a23641aa77bc9c/debian-openssl-3.0.x/libquery_engine.so.node.gz.sha256 --output libquery_engine.so.node.gz.sha256
```
```
wget https://binaries.prisma.sh/all_commits/d9a4c5988f480fa576d43970d5a23641aa77bc9c/debian-openssl-3.0.x/libquery_engine.so.node.gz.sha256
```

**Reply by BEDIRHANSAMSA** [reply]
![image](https://github.com/prisma/prisma/assets/44173331/7f3006f4-dc29-406d-8614-a874538bfaac)

**Reply by ludralph** [reply]
🤔 there could be a number of factors for this behaviour, you might want to look at this [post](https://stackoverflow.com/questions/20986395/why-does-curl-not-work-but-wget-works)

**Reply by BEDIRHANSAMSA** [reply]
Could it have something to do with ipv6?

**Reply by BEDIRHANSAMSA** [reply]
Yesss, when I type this command its work
```curl
wget --inet4-only https://binaries.prisma.sh/all_commits/d9a4c5988f480fa576d43970d5a23641aa77bc9c/debian-openssl-3.0.x/libquery_engine.so.node.gz.sha256
```

but how i fix?

**Reply by BEDIRHANSAMSA** [reply]
i disabled ipv6 and its works. but why ipv6 not working?


how to disable: https://itsfoss.com/disable-ipv6-ubuntu-linux/

**Reply by ludralph** [reply]
This could also arise due to router configuation issues, incorrect DNS sever address or ipv6 compatibility issues.

**Reply by janpio** [reply]
We recently changed some configuration of binaries.prisma.sh. We are currently investigating if that could have had impact on the IPv6-only availability of our engine files - which should not be the case. We'll fix it if that is currently the case.

**Reply by janpio** [reply]
We are pretty confident there is no problem in our configuration: https://dnschecker.org/#AAAA/binaries.prisma.sh

Can you please try again with IPv6 enabled?

**Reply by BEDIRHANSAMSA** [reply]
When i try only ipv6 gives error
![image](https://github.com/prisma/prisma/assets/44173331/963f8ad0-4acf-48e9-b368-b063a0948ff3)

**Reply by janpio** [reply]
Does it still work with curl? What about Prisma CLI?

We have no other reports that it is not working for anyone, so we are _pretty_ certain this is a configuration thing on your machine or network, not a general problem for all our users.
